migrations/20200409232153_rewrite_twitter_urls.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+use Phinx\Migration\AbstractMigration;
+
+class RewriteTwitterUrls extends AbstractMigration
+{
+    /**
+     * Change Method.
+     *
+     * Write your reversible migrations using this method.
+     *
+     * More information on writing migrations is available here:
+     * http://docs.phinx.org/en/latest/migrations.html#the-abstractmigration-class
+     *
+     * The following commands can be used in this method and Phinx will
+     * automatically reverse them when rolling back:
+     *
+     *    createTable
+     *    renameTable
+     *    addColumn
+     *    renameColumn
+     *    addIndex
+     *    addForeignKey
+     *
+     * Remember to call "create()" or "update()" and NOT "save()" when working
+     * with the Table class.
+     */
+    public function up()
+    {
+        $sql = "UPDATE messages 
+                INNER JOIN contacts on contacts.id = messages.contact_id
+                SET messages.message = REPLACE(messages.message, concat('https://twitter.com/statuses/', messages.data_source_message_id), concat('https://twitter.com/', contacts.contact, '/status/', messages.data_source_message_id))
```

Comment:
I thought this couldn't be done, because when I was checking a database with tweet messages, I couldn't see the tweet user handles anywhere. Turns out that we are storing the numeric ID for the user.

It also turns out that twitter *at this point in time now* will accept that numeric ID in the tweet URL  (i.e. https://twitter.com/:userNumId/status/:tweetId ) and redirect to the URL with the user handle.

So that's good now, it makes me queasy though 🤷  Should we start storing the screen name, should we start storing both? Big questions for a Friday